### PR TITLE
Jg shelter crud incomplete flash us29

### PR DESCRIPTION
**Pull request by elguapogordo**
added tests for
  * flash message for any/each fields left empty when creating a shelter
  * flash message for any/each fields left empty when updating a shelter
added multiple flash message function for
  * creating a shelter
  * updating a shelter

### CODE REVIEW

File: app/controllers/shelters_controller.rb [hunk 1]
```diff
@@ -13,8 +13,17 @@ def new
...
+    shelter = Shelter.new(shelter_params)
+    if shelter.save
+      redirect_to '/'
+    else
+      flash[:error] = ""
+      redirect_back(fallback_location: root_path)
+      missing(shelter_params).each do |param|
+        flash[:error] += "The #{param} field must not be left blank. "
+      end
+      flash[:error]
```
Comment (elguapogordo) [inline app/controllers/shelters_controller.rb:25]: this allows multiple messages to be concatenated into one flash message

File: spec/features/shelters/shelter_crud_incomplete_flash_spec.rb [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+      click_on 'Create Shelter'
+
+      expect(page).to have_content("The zip field must not be left blank")
+
+      visit "/shelters/new"
+
+      click_on 'Create Shelter'
+
+      expect(page).to have_content("The name field must not be left blank")
```
Comment (tony-gomes) [inline spec/features/shelters/shelter_crud_incomplete_flash_spec.rb:74]: This reminds me that I need to brush up on how to use iteration in rspec...

File: app/controllers/shelters_controller.rb [hunk 3]
```diff
@@ -28,9 +37,16 @@ def edit
...
-    shelter.update(shelter_params)
-
-    redirect_to "/shelters/#{shelter.id}/"
+    if shelter.update(shelter_params)
+      redirect_to "/shelters/#{shelter.id}/"
+    else
+      flash[:error] = ""
+      redirect_back(fallback_location: root_path)
+      missing(shelter_params).each do |param|
+        flash[:error] += "The #{param} field must not be left blank. "
```
Comment (tony-gomes) [inline app/controllers/shelters_controller.rb:46]: This is dope.

File: app/controllers/shelters_controller.rb [hunk 4]
```diff
@@ -44,4 +60,10 @@ def pending_pets
     end
   end
 
+  def missing(parameters)
+    parameters.keys.find_all do |key|
```
Comment (tony-gomes) [inline app/controllers/shelters_controller.rb:64]: sign of a Rubyist.

File: spec/features/shelters/shelter_crud_incomplete_flash_spec.rb [hunk 5]
```diff
@@ -0,0 +1,140 @@
+require 'rails_helper'
```
Comment (mikez321) [inline spec/features/shelters/shelter_crud_incomplete_flash_spec.rb:1]: Quite the test here

**Pull request merged**